flight/states/takeoff.py

```diff
@@ -47,3 +50,60 @@ class Takeoff(State):
             altitude: float = round(position.relative_altitude_m, 2)
             if altitude >= 2:
                 return True
+
+    async def takeoff(self, drone: System) -> None:
+
+        logging.info("Taking off towards first pylon")
+
+        up_speed: float = -2
+
+        async for gps in drone.telemetry.position():
+
+            # Find the current latitude and longitude positions
+            curr_lat = round(gps.latitude_deg, 8)
+            curr_lon = round(gps.longitude_deg, 8)
+
+            x = ((target_lon - curr_lon)* 40000 * math.cos((target_lat + curr_lat) * math.pi / 360) / 360) * 1000
+
+            y = ((target_lat - curr_lat) * 40000 / 360) * 1000
+
+            # Calculate the degree to point the drone at
+            try:
+                deg = round((((math.atan(x / y) / math.pi) * 180)))
+
+                if y < 0:
+                    z = 180
+                    z = math.copysign(z, deg)
+                    deg = z + deg
+            except ZeroDivisionError:
+                deg = round((((math.asin(x / (math.sqrt((x ** 2) + (y ** 2)))) / math.pi)* 180)))
+
+                if y < 0:
+                    z = 180
+                    z = math.copysign(z, deg)
+                    deg = z + deg
+
+            # Calculate the needed velocity needed to reach the target lat lon points
+            try:
+                dx = math.copysign(35 * math.cos(math.atan(y / x)), x)
+                dy = math.copysign(35 * math.sin(math.atan(y / x)), y)
+
+            except ZeroDivisionError:
+                dx = math.copysign(35 * math.cos(math.asin(y / (math.sqrt((x ** 2) + (y ** 2))))), x)
+                dy = math.copysign(35 * math.sin(math.asin(y / (math.sqrt((x ** 2) + (y ** 2))))), y)
+
+            # Start the drone pointing in the direction of the first pylon
+            await drone.offboard.set_velocity_ned(
+                sdk.VelocityNedYaw(dy, dx, up_speed, deg)
+            )
+
+            # Loops until the desired altitude has been attained,
+            # then sets the upward velocity to 0 and returns
+            if await self.wait_alt(drone) == True:
```

Comment:
if wait_alt doesnt return True, then the drone will be in pieces, so dont use it as an if statement, just say
```python
await self.wait_alt(drone)
other_code_that_happens_after()
```